### ISSUE TITLE
Clarify usage guidelines / licensing for code examples in the tutorials/classref

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot-docs#2772]
While the documentation's text overall is covered by a CC-BY 3.0 license, which means that proper attribution should be given when reusing its content, we should clarify whether this applies to code examples given in tutorials and the class reference.

As it is right now, the license is the same for everything, so for small snippets one could say that fair use means no attribution needed, but for longer (e.g. 50 lines) scripts it's a bit tougher.

IMO we should specify in the About section and in the copyright headers that all code blocks in the documentation are to be considered under Public Domain (or CC0 where PD does not apply), so that users can copy/paste documentation examples without having to bother with giving attribution.

WDYT?

**Comment by Calinou** [comment]
I agree with licensing code samples under CC0, though we will have to ask people who contributed subtantial code examples.

**Comment by Vennnot** [comment]
If the code examples exist publicly in the documentation, it'll be difficult to chase down those who contribute it all the code. I'd vote for making all of the code public domain right now.
